internal/extender/overhead.go

```diff
@@ -176,18 +219,24 @@ func podToResources(ctx context.Context, pod *v1.Pod) *resources.Resources {
 	return res
 }
 
-// GetOverhead fills overhead information for given nodes, and falls back to the median overhead
-// of the instance group if the node is not found
-func (o OverheadComputer) GetOverhead(ctx context.Context, nodes []*v1.Node) resources.NodeGroupResources {
+func (o *OverheadComputer) getPodNodeInstanceGroup(pod *v1.Pod) (string, error) {
+	node, err := o.nodeLister.Get(pod.Spec.NodeName)
+	if err != nil {
+		return "", werror.Wrap(err, "node does not exist in cache", werror.SafeParam("nodeName", pod.Spec.NodeName))
+	}
+	return node.Labels[o.instanceGroupLabel], nil
+}
+
+func (o OverheadComputer) getOverheadByNode(ctx context.Context, overhead Overhead, nodes []*v1.Node) resources.NodeGroupResources {
 	o.overheadLock.RLock()
```

Comment:
I would prefer using the lock at call site, it will make explicit that `overheadLock` guards the overhead.